guake/guake_app.py

```diff
@@ -101,14 +101,25 @@ class Guake(SimpleGladeApp):
     def __init__(self):
         def load_schema():
             log.info("Loading Gnome schema from: %s", SCHEMA_DIR)
+            with open(os.path.join(SCHEMA_DIR, "version"), encoding="utf-8") as version:
+                if version.read().strip() != guake_version():
+                    raise ValueError("Old Guake schema found")
+
             return Gio.SettingsSchemaSource.new_from_directory(
                 SCHEMA_DIR, Gio.SettingsSchemaSource.get_default(), False
             )
 
         try:
             schema_source = load_schema()
-        except GLib.Error:  # pylint: disable=catching-non-exception
-            log.exception("Unable to load the GLib schema, try to compile it")
+        except (
+            GLib.Error,
```

Comment:
Prefer:

```
try:
 pass
except GLib.Error:
 log.exception(...)
except (FileNotFoundError, ValueError):
 log.exception(...)
```

to reduce `isinstance()` compare

Reply:
I did actually do that at first, then I rearranged it to the way it is because both cases have to run the next two lines and I didn't want to copy/paste code.

Reply:
I didn't catch next two line :p 

then how about try-except-finally?

```
try
...
except A:
...
except B:
...
finally:
  try_to_compile...
  schema_source = ...
```

Reply:
The finally part of try-except always runs, even when there is no exception. This is not quite the behavior we want in this scenario.

Reply:
you are right, just leave it like this :)